back_end/src/apis/prerequisites.py

```diff
@@ -0,0 +1,93 @@
+from flask_cors import CORS
+from flask import Blueprint, request, jsonify
+
+from ..models.prerequisites import Prerequisite
+
+prereq_api_bp = Blueprint('prereq_api', __name__)
+CORS(prereq_api_bp, supports_credentials=True)
+
+# db.create_all()
+# db.session.commit()
+
+'''
+@author: Jiazheng Liu
+
+possible other methods:
+get classes that has the given class as prereq
+check if the given prereq is enough for a class
+'''
+
+@prereq_api_bp.route('/create_class_with_prereq', methods=['POST'])
+def create_class_with_prereq():
+    '''
+    Route to create a user
+    input   class_id(int), required_class(str, i.e. "CSE30, CSE100")
+    output  'class with prereq created' or 'class with prereq existed'
+
+    @author: Jiazheng Liu
+    '''
+    req_data = request.get_json()
+    class_id = req_data.get('class_id')
+    required_classes = req_data.get('required_classes')
+    status = Prerequisite.create_prereq(class_id = class_id, required_classes = required_classes)
+    if status:
+        return jsonify({'reason': 'class with prereq created'}), 200
+    else:
+        return jsonify({'reason': 'class with prereq existed'}), 300
+
+
+
+@prereq_api_bp.route('/get_all_classes_with_prereq', methods=['GET'])
+def get_all_classes_with_prereq():
+    '''
+    Route to get the prereq for all classes
+    input   None
+    output  prereq for all classses
+
+    @author: Jiazheng Liu
+    '''
+    clss = Prerequisite.get_prereqs()
+    return jsonify({'reason': 'success', 'result': clss}), 200
+
+# FIXME:
+# If the user wants to show prereq graph, and the frontend needs the prereq for a list of classes,
+# do they use a for loop and call the api that gets only one class at a time, or do we need an api
+# that takes a list and return a list of prereq (but how the list is passed? Through URL?)
+@prereq_api_bp.route('/get_class_with_prereq', methods=['GET'])
+def get_class_with_prereq():
+    '''
+    Route to get the prereq for one class
+    input   class_id(int)
+    output  prereq for the given class if the class exists in the BD or
+            'failed: class with prereq DNE'
+
+    @author: Jiazheng Liu
+    '''
+    # clss = Prerequisite.get_prereqs()
+    class_id = request.args.get('class_id')
```

Comment:
To answer your question, they can pass in a ";" separated string indicating they want to access the prereq of multiple classes. In this case, you should do:
```class_id = request.args.get('class_id')
class_id_list = class_id.split(';')
class_id_list = list(map(lambda: x int(x)), class_id_list)
```
Then in the API we loop through all the classes that they send. So I think we should write an API like this.

However, this is not suitable for prereq graph, because we need to calculate a group on the server and send the graph directly to them. I haven't figured out how to do that and I will look into it further.